queries/migrations.create_prune_galleries_trigger.sql

```diff
@@ -11,17 +11,50 @@ SELECT g.user_id INTO _user_id FROM galleries g
                                         JOIN wallets w ON g.user_id = w.user_id AND w.id = OLD.wallet_id
     LIMIT 1;
 
--- Cleanup pivot table
+-- Mark all gallery nfts as soft deleted if no longer owned by any user
+WITH unassinged_nfts AS (
+    SELECT gallery_id, nft_id FROM galleries
+                                       JOIN nft_gallery ng on galleries.id = ng.gallery_id
+                                       JOIN nfts n on ng.nft_id = n.id AND n.wallet_id IS NULL
+    WHERE user_id = _user_id
+)
+UPDATE nft_gallery
+    SET deleted_at = NOW()
+FROM unassinged_nfts un
+WHERE nft_gallery.gallery_id = un.gallery_id AND nft_gallery.nft_id = un.nft_id;
+
```

Comment:
Since this trigger is executed per nft row, we do not need to select all nfts each time. It takes 1-2 seconds for me on a `composer db:live` table when removing the wallet for all nfts at once.

We should be able to simplify this to:

```sql
UPDATE nft_gallery
SET deleted_at = NOW()
WHERE nft_gallery.nft_id = NEW.id AND NEW.wallet_id IS NULL;
```

Didn't notice any functional difference, except that it makes it instant.

Reply:
@oXtxNt9U updated ✅